### PR TITLE
[feat] Create production flag on the backend

### PR DESCRIPTION
**Pull request by ptisserand**
- [x] issue #3 
- [x] follows contribution [guide](https://github.com/keep-starknet-strange/art-peace/blob/main/CONTRIBUTING.md)
- [ ] code change includes tests
- [ ] breaking change

A `--production` flag on the backend to disable `devnet` related routes.

### CODE REVIEW

**Comment by ptisserand** [comment]
Hi @b-j-roberts , do you think I should try to use `context` instead of this really simple implementation ?

**Comment by b-j-roberts** [comment]
> Hi @b-j-roberts , do you think I should try to use `context` instead of this really simple implementation ?

Thought about this some. I think the best option would be to add the "production" option to the `backendConfig` which are passed from the backend `.config.json` files here https://github.com/keep-starknet-strange/art-peace/tree/main/configs 

Then the production flag will act as an override of that config option, which will be defaulted to false.

You might need to swap the order of `InitRoutes` and `NewBackend` in `main.go` so the global variables are setup.
Hope that makes sense

**Comment by ptisserand** [comment]
Hi @b-j-roberts I have updated the PR according to your last comments.

Should I also add `production` in JSON config files?

File: backend/routes/templates.go [hunk 1]
```diff
@@ -93,7 +93,12 @@ func addTemplateData(w http.ResponseWriter, r *http.Request) {
 }
 
 func addTemplateHashDevnet(w http.ResponseWriter, r *http.Request) {
-	// TODO: Disable this in production
+	// Disable this in production
+	if core.ArtPeaceBackend.BackendConfig.Production {
```
Comment (b-j-roberts) [inline backend/routes/templates.go:97]: This is good, but could you also disable the respective routes in the `Init*Routes` functions so they are not callable?

File: backend/main.go [hunk 2]
```diff
@@ -29,11 +41,16 @@ func main() {
 		panic(err)
 	}
 
+	if isFlagSet("production") {
```
Comment (b-j-roberts) [inline backend/main.go:44]:
```suggestion
	if *production {
```

I believe you can just do this, and remove the `isFlagSet` function.

**Review by b-j-roberts — CHANGES_REQUESTED** [review]

File: backend/routes/templates.go (hunk 1 above)
Comment (ptisserand) [reply backend/routes/templates.go:99]: Ok, handles for `/addTemplateHashDevnet` and `/placePixelDevnet` are enabled only when production flag is false.

File: backend/main.go (hunk 2 above)
Comment (ptisserand) [reply backend/main.go:44]: It's not the same behavior.
With `isFlagSet`  and `--production=false` we can force the production mode to be `false` even if is set to `true` in configuration file.

**Review by b-j-roberts — APPROVED** [review]
Nice work!

**Pull request merged**